Mathlib/Analysis/Convex/Deriv.lean

```diff
@@ -365,19 +365,221 @@ lemma ConvexOn.slope_mono (hfc : ConvexOn 𝕜 s f) (hx : x ∈ s) : MonotoneOn
   (slope_fun_def_field f _).symm ▸ fun _ hy _ hz hz' ↦ hfc.secant_mono hx (mem_of_mem_diff hy)
     (mem_of_mem_diff hz) (not_mem_of_mem_diff hy :) (not_mem_of_mem_diff hz :) hz'
 
+lemma ConvexOn.monotoneOn_slope_gt (hfc : ConvexOn 𝕜 s f) (hxs : x ∈ s) :
+    MonotoneOn (slope f x) {y ∈ s | x < y} :=
+  (hfc.slope_mono hxs).mono fun _ ⟨h1, h2⟩ ↦ ⟨h1, h2.ne'⟩
+
+lemma ConvexOn.monotoneOn_slope_lt (hfc : ConvexOn 𝕜 s f) (hxs : x ∈ s) :
+    MonotoneOn (slope f x) {y ∈ s | y < x} :=
+  (hfc.slope_mono hxs).mono fun _ ⟨h1, h2⟩ ↦ ⟨h1, h2.ne⟩
+
 /-- If `f : 𝕜 → 𝕜` is concave on `s`, then for any point `x ∈ s` the slope of the secant line of `f`
 through `x` is antitone on `s \ {x}`. -/
 lemma ConcaveOn.slope_anti (hfc : ConcaveOn 𝕜 s f) (hx : x ∈ s) :
     AntitoneOn (slope f x) (s \ {x}) := by
   rw [← neg_neg f, slope_neg_fun]
   exact (ConvexOn.slope_mono hfc.neg hx).neg
 
+lemma ConcaveOn.antitoneOn_slope_gt (hfc : ConcaveOn 𝕜 s f) (hxs : x ∈ s) :
+    AntitoneOn (slope f x) {y ∈ s | x < y} :=
+  (hfc.slope_anti hxs).mono fun _ ⟨h1, h2⟩ ↦ ⟨h1, h2.ne'⟩
+
+lemma ConcaveOn.antitoneOn_slope_lt (hfc : ConcaveOn 𝕜 s f) (hxs : x ∈ s) :
+    AntitoneOn (slope f x) {y ∈ s | y < x} :=
+  (hfc.slope_anti hxs).mono fun _ ⟨h1, h2⟩ ↦ ⟨h1, h2.ne⟩
+
+variable [TopologicalSpace 𝕜] [OrderTopology 𝕜]
+
+lemma bddBelow_slope_Ioi_of_mem_interior (hfc : ConvexOn 𝕜 s f) (hxs : x ∈ interior s) :
+    BddBelow (slope f x '' {y ∈ s | x < y}) := by
```

Comment:
There is no `Ioi` in the statement

Reply:
Yeah that was meant to refer to `{y ∈ s | x < y}`, but that's not really Ioi because of the `y ∈ s` condition.

Reply:
You could write it as `s ∩ Ioi x`, I guess